### PR TITLE
{Packaging} Trim mgmt SDKs

### PR DESCRIPTION
**Pull request by jiasli**
**Description**

Reduce Azure CLI's package size per

- https://github.com/Azure/azure-cli/issues/7387

Add `trim-sdk.py` script to trim Python SDKs by

- Removing `aio` folders
- Removing unused API versions

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 1]
```diff
@@ -417,6 +417,29 @@ def default_api_version(self):
...
+        # src/azure-cli/azure/cli/command_modules/acr/_client_factory.py:8
+        'VERSION_2019_05_01_PREVIEW': "2019-05-01-preview",
+        'VERSION_2019_06_01_PREVIEW': "2019-06-01-preview",
+        'VERSION_2020_11_01_PREVIEW': "2020-11-01-preview",
+        'VERSION_2021_08_01_PREVIEW': "2021-08-01-preview",
+        'VERSION_2022_02_01_PREVIEW': "2022-02-01-preview",
+    },
+    ResourceType.MGMT_CONTAINERSERVICE: {
+        # src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_custom.py:50
+        'ManagedClusterAddonProfile': '2020-03-01',
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:442]: Ad-hoc API versions should be removed: 
- https://github.com/Azure/azure-cli/issues/23389

**Comment by jiasli** [comment]
**Build Windows MSI** task in CI has been reduced from [16m 54s](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1861365&view=logs&j=78898306-33c5-5664-2f1b-a3010f777c49) to [11m 38s](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1861435&view=logs&j=78898306-33c5-5664-2f1b-a3010f777c49).

**Review by necusjz — APPROVED** [review]

File: scripts/trim_sdk.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        _rmtree(full_path)
+
+
+def _print_folder_size(folder):
+    size, count = calculate_folder_size(folder)
+    size_in_mb = size / 1048576
+    _LOGGER.info(f"{size_in_mb:.2f} MB, {count} files")
+
+
+def _get_all_sdks_to_trim():
```
Comment (jsntcy) [inline scripts/trim_sdk.py:111]: It seems this function is not used, do we need it?

Comment (jiasli) [reply scripts/trim_sdk.py:111]: No. It's not. I keep it there so that we can trim all SDKs in the future.

File: scripts/trim_sdk.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    _LOGGER.info(f'API folders to remove: {remove_api_folders}')
+
+    for api_folder in remove_api_folders:
+        full_path = os.path.join(sdk_path, api_folder)
+        _rmtree(full_path)
+
+
+def _print_folder_size(folder):
+    size, count = calculate_folder_size(folder)
+    size_in_mb = size / 1048576
```
Comment (zhoxing-ms) [inline scripts/trim_sdk.py:107]: We may need to make this number `1048576` more readable, such as `1048576` --> `1024*1024`

Comment (jiasli) [inline scripts/trim_sdk.py:107]:
```suggestion
    size_in_mb = size / 1048576  # 1 MB = 1024 * 1024 B = 1048576 B
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**